Tests/ApolloTests/ReadFieldValueTests.swift

```diff
@@ -105,19 +99,13 @@ class ReadFieldValueTests: XCTestCase {
     XCTAssertNil(value)
   }
   
-  func testGetOptionalScalarWithWrongType() throws {
+  func testGetOptionalScalarWithDifferentType() throws {
     let object: JSONObject = ["name": 10]
     let field = GraphQLField("name", type: .scalar(String.self))
     
-    XCTAssertThrowsError(try readFieldValue(field, from: object)) { (error) in
-      if let error = error as? GraphQLResultError, case JSONDecodingError.couldNotConvert(let value, let expectedType) = error.underlying {
-        XCTAssertEqual(error.path, ["name"])
-        XCTAssertEqual(value as? Int, 10)
-        XCTAssertTrue(expectedType == String.self)
-      } else {
-        XCTFail("Unexpected error: \(error)")
-      }
-    }
+    let value = try readFieldValue(field, from: object) as! String
```

Comment:
I would strongly recommend against force-unwrapping here as instead of failing the test, it crashes the whole test suite. Usually I do something like 

```swift
guard let foo = bar as? String else {
   XCTFail("Wrong type!")
   return 
}

XCTAssertEqual(foo, "expectedFoo")
```

Reply:
Totally agree. Force unwrapping here is a non go. Your solution sounds very clean to me. I will update this code too.